### PR TITLE
Palmira.benavente/implementation dialog fragment

### PR DESCRIPTION
**Pull request by PalmiraBenavente**
- Dialog fragment implementation: Added contract, model, view and presenter for the dialog fragment.
- Added layout, styles, dimensions, color, strings for DialogFragment.

image
image
image

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    private ListenerDialogFragment listenerDialogFragment;
+
+    public DialogSpaceParkingPresenter(DialogSpaceParkingContract.DialogFragmentView view, ListenerDialogFragment listenerDialogFragment){
+        this.view=view;
+        this.listenerDialogFragment=listenerDialogFragment;
+    }
+
+    @Override
+    public void onButtonPressed() {
+        String str = view.getSpacesFromEditText();
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java]: rename the variable please

File: app/src/main/res/layout/fragment_dialog_space_parking.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    android:background="@color/medium_aqua"
+    >
+    <TextView
+        android:id="@+id/text_view_dialog_fragment_spaces"
+        android:text="@string/text_view_dialog_fragment_set_space"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        style="@style/TextFragmentViewStyle"
+        android:layout_margin="@dimen/margin_dialog_fragment_size_xbig"
+        />
```
Comment (mauroLonghi) [inline app/src/main/res/layout/fragment_dialog_space_parking.xml]: pls put the end tag in the previous sentences

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.example.parkingsystem.mvp.presenter;
+
+import com.example.parkingsystem.mvp.contract.DialogSpaceParkingContract;
+import com.example.parkingsystem.listener.ListenerDialogFragment;
+
+public class DialogSpaceParkingPresenter implements DialogSpaceParkingContract.DialogFragmentPresenter {
+    private DialogSpaceParkingContract.DialogFragmentView view;
+    private ListenerDialogFragment listenerDialogFragment;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java]: do not save the listener as an instance variable.. just pass it as a parameter when you need it

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 4]
```diff
@@ -13,7 +13,12 @@ public ParkingPresenter(ParkingContract.MainActivityModel model, ParkingContract
...
     public void onSelectParkingButtonPressed() {
-        model.setParkingSpace();
-        view.showParkingSpacesAvailable(model.getParkingSpace());
+        view.showSelectionParkingSpaces(model.getParkingSpace());
+    }
+
+    @Override
+    public void onSetParkingButtonPressed(int spaces) {
+        model.setParkingSpace(spaces);
+        view.toastShowSpaces(spaces);
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: you should get the info from the model
`        view.toastShowSpaces(model.getSpaces());`

File: app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package com.example.parkingsystem.mvp.view;
+
+import com.example.parkingsystem.SpacesParkingDialogFragment;
+import com.example.parkingsystem.databinding.FragmentDialogSpaceParkingBinding;
+import com.example.parkingsystem.mvp.contract.DialogSpaceParkingContract;
+import java.lang.ref.WeakReference;
+
+public class DialogSpaceParkingView implements DialogSpaceParkingContract.DialogFragmentView {
+    private FragmentDialogSpaceParkingBinding binding;
+    private WeakReference<SpacesParkingDialogFragment> dialogFragmentWeakRef;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java]: create a FragmentView in the base package for the weakReferences and the general things, just like the ActivityView

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java [hunk 6]
```diff
@@ -17,10 +18,13 @@ public ParkingView(Activity activity, ActivityMainBinding binding) {
...
-            Toast.makeText(context, context.getString(R.string.toast_main_activity_select_space_parking, parkingSpace), Toast.LENGTH_LONG).show();
-        }
+    public void showSelectionParkingSpaces(int parkingSpace) {
+        FragmentManager fragmentManager = getFragmentManager();
+        SpacesParkingDialogFragment spacesParkingDialogFragment = SpacesParkingDialogFragment.newInstance();
+        spacesParkingDialogFragment.show(fragmentManager,"dialog_fragment");
+    }
+
+    public void toastShowSpaces(int parkingSpace){
+        Toast.makeText(getContext(), getContext().getString(R.string.toast_main_activity_select_space_parking, parkingSpace),Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: please, validate that the getContext is not null

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/SpacesParkingDialogFragment.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    private FragmentDialogSpaceParkingBinding binding;
+    private DialogSpaceParkingContract.DialogFragmentPresenter presenter;
+    private ListenerDialogFragment listenerDialogFragment;
+
+    public SpacesParkingDialogFragment() {}
+
+    public static SpacesParkingDialogFragment newInstance() {
+        SpacesParkingDialogFragment dialogFragment = new SpacesParkingDialogFragment();
+        Bundle bumble = new Bundle();
+        dialogFragment.setArguments(bumble);
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystem/SpacesParkingDialogFragment.java]: this is not needed. Dont make sense to send an empty bundle

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java (hunk 1 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java]: Done

File: app/src/main/res/layout/fragment_dialog_space_parking.xml (hunk 2 above)
Comment (PalmiraBenavente) [reply app/src/main/res/layout/fragment_dialog_space_parking.xml]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java (hunk 3 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 4 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java (hunk 5 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java (hunk 6 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: Done

File: app/src/main/java/com/example/parkingsystem/SpacesParkingDialogFragment.java (hunk 7 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/SpacesParkingDialogFragment.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        this.binding = binding;
+    }
+
+    public String getSpacesFromEditText() {
+        return binding.editTextDialogFragmentSpaces.getText().toString();
+    }
+
+    @Override
+    public void dismissDialogFragment() {
+        SpacesParkingDialogFragment dialogFragmentReference = getParkingFragment();
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java]: here you should validate that the dialogFragmentReference is not null before using it

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java [hunk 9]
```diff
@@ -17,10 +18,14 @@ public ParkingView(Activity activity, ActivityMainBinding binding) {
...
-        }
+    public void showSelectionParkingSpaces(int parkingSpace) {
+        FragmentManager fragmentManager = getFragmentManager();
+        SpacesParkingDialogFragment spacesParkingDialogFragment = SpacesParkingDialogFragment.newInstance();
+        spacesParkingDialogFragment.show(fragmentManager, "dialog_fragment");
+    }
+
+    public void toastShowSpaces(int parkingSpace) {
+        if (getContext() != null)
+            Toast.makeText(getContext(), getContext().getString(R.string.toast_main_activity_select_space_parking, parkingSpace), Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: do not call again getContext() just save the context into a variable and use the variable. If you do it like this the getContext() can still be null

File: app/src/main/java/com/example/parkingsystem/mvp/view/base/FragmentView.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package com.example.parkingsystem.mvp.view.base;
+
+import com.example.parkingsystem.SpacesParkingDialogFragment;
+import java.lang.ref.WeakReference;
+
+public class FragmentView {
+    private WeakReference<SpacesParkingDialogFragment> dialogFragmentWeakRef;
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystem/mvp/view/base/FragmentView.java]: this should be Fragment type

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java (hunk 8 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/DialogSpaceParkingView.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java (hunk 9 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ParkingView.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/base/FragmentView.java (hunk 10 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/base/FragmentView.java]: Done

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java (hunk 1 above)
Comment (mauroLonghi) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/DialogSpaceParkingPresenter.java]: still here

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**